grommet/grommet
# Update theme styling options for DataTable header

#### What does this PR do?
This PR extends theming capabilities of data table header cells. The contents of this cell can include a label (clickable if column is sortable, otherwise just text/custom render), a search button (not encouraged but already exists and left untouched), and a resizer. The `theme.dataTable.header` structure now supports:
```
         background: undefined,
         border: undefined,
         font: {
           weight: undefined,
           size: undefined,
         },
        gap: 'small', // previously defined inline, but now giving users to style via theme and setting inline value as default
         hover: {
           background: undefined,
        },
        pad: undefined,
```
`background` and `border` will apply to the overall `th`, but the remaining will affect the styling of the content instead. In the case that the content is just a Text, then it will be applied to the Box surrounding the Text. In the case that the content is sortable and becomes a Button, then the theming will be applied to that Button. Additionally, the column label (whether Text or Button) now has `flex="grow"` which allows it to take up any remaining space in the column.

#### Where should the reviewer start?
src/js/components/DataTable/Header.js

#### What testing has been done on this PR?
Added a jest test. 

#### How should this be manually tested?
In storybook for DataTable Sort, apply the following theme locally:
```
const customTheme = {
      dataTable: {
        header: {
          background: 'skyblue',
          border: {
            color: 'green',
            size: 'medium',
          },
          gap: 'none',
          pad: { horizontal: 'small', vertical: 'xsmall' },
          font: {
            weight: 'bold',
          },
          hover: {
            background: {
              color: 'light-2',
            },
          },
        },
      },
    };
```
You should see the following as a result (the grey background was where the mouse was hovered over):
<img width="1680" alt="Screen Shot 2020-09-22 at 4 01 35 PM" src="https://user-images.githubusercontent.com/12522275/93946414-0e446200-fcee-11ea-9a3d-0db8b852f4c9.png">

#### Any background context you want to provide?

#### What are the relevant issues?
Closes https://github.com/grommet/grommet/issues/4454

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
Yes, updated here.

#### Should this PR be mentioned in the release notes?
Yes.

#### Is this change backwards compatible or is it a breaking change?
Yes, backwards compatible.
